src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsWriter.java

```diff
@@ -277,11 +279,24 @@ private int getLiveDocs(KNNVectorValues<?> vectorValues) throws IOException {
         return liveDocs;
     }
 
-    private void initQuantizationStateWriterIfNecessary() throws IOException {
+    private void writeQuantizationState(QuantizationParams quantizationParams, QuantizationState quantizationState, int fieldNumber)
+        throws IOException {
+
+        // We will not be writing quantization state for 8 bits into a segment file because we are not quantizing the query vectors and
```

Comment:
```suggestion
        // We will not write the 8-bit quantization state into a segment file because query vectors are not being quantized. Additionally, the template index trained in the quantization state will not be stored, as it is intended for later use in the index build strategy during data ingestion.
```